File: app/src/main/java/ua/hospes/rtm/ui/race/widgets/CustomToggleButton.java
package ua.hospes.rtm.ui.race.widgets;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.ToggleButton;

/**
 * @author Andrew Khloponin
 */
public class CustomToggleButton extends ToggleButton {
    private final long minimumInterval = 5000;
    private long previousTimestampClick = System.currentTimeMillis();


    public CustomToggleButton(Context context) {
        super(context);
    }

    public CustomToggleButton(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public CustomToggleButton(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public CustomToggleButton(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    @Override
    public boolean performClick() {
//        long currentTimestamp = System.currentTimeMillis();

//        if (currentTimestamp - previousTimestampClick > minimumInterval) {
//            previousTimestampClick = currentTimestamp;
            return super.performClick();
//        }
//        return false;
    }
}

File: app/src/main/java/ua/hospes/rtm/widgets/CustomToggleButton.kt
package ua.hospes.rtm.widgets

import android.content.Context
import android.util.AttributeSet
import android.widget.ToggleButton

/**
 * @author Andrew Khloponin
 */
class CustomToggleButton : ToggleButton {
    private val minimumInterval: Long = 5000
    private val previousTimestampClick = System.currentTimeMillis()


    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {}

    override fun performClick(): Boolean {
        //        long currentTimestamp = System.currentTimeMillis();

        //        if (currentTimestamp - previousTimestampClick > minimumInterval) {
        //            previousTimestampClick = currentTimestamp;
        return super.performClick()
        //        }
        //        return false;
    }
}